File: src/multi_subscribe.cpp
# include<ros/ros.h>
# include<std_msgs/String.h>
# include<std_msgs/Float32.h>
# include<std_msgs/Float32MultiArray.h>

void callback1(const std_msgs::StringConstPtr& str){
    std::cout<<"received str:"<<str->data<<std::endl;
}

void callback2(const std_msgs::Float32ConstPtr& flt){
    std::cout<<"received float:"<<flt->data<<std::endl;
}

void callback3(const std_msgs::Float32MultiArrayConstPtr& arr){
    std::cout<<"received array:";
    for (size_t i = 0; i < arr->data.size(); i++)
    {
        std::cout<<arr->data[i];
    }
    std::cout<<std::endl;
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "multi_subscribe");
    ros::NodeHandle nh;

    ros::Subscriber sub1 = nh.subscribe("topic_str",1,callback1);
    ros::Subscriber sub2 = nh.subscribe("topic_flt",1,callback2);
    ros::Subscriber sub3 = nh.subscribe("topic_arr",1,callback3);
    ros::spin();
    return 0;
}

File: scripts/multi_subscribe.py
# coding=utf-8
import rospy
from std_msgs.msg import String
from std_msgs.msg import Float32
from std_msgs.msg import Float32MultiArray


def callback1(str):
    print "received str:", str.data


def callback2(flt):
    print "received flt:", flt.data


def callback3(arr):
    print "received arr:", arr.data


if __name__ == "__main__":
    rospy.init_node("multi_subscribe")

    sub1 = rospy.Subscriber("topic_str", String, callback1)
    sub2 = rospy.Subscriber("topic_flt", Float32, callback2)
    sub3 = rospy.Subscriber("topic_arr", Float32MultiArray, callback3)

    rospy.spin()
